code/modules/reagents/reagent_containers/blood_pack.dm

```diff
@@ -85,21 +120,107 @@
 	blood_type = "O-"
 
 /obj/item/weapon/reagent_containers/blood/empty
-	name = "\improper Empty Bloodpack"
+	name = "\improper empty bloodpack"
 	desc = "Seems pretty useless... Maybe if there were a way to fill it?"
-	icon_state = "empty"
+	icon_state = "bloodpack"
 	New()
 		..()
 		blood_type = null
 		reagents.clear_reagents()
 		update_icon()
 
 /obj/item/weapon/reagent_containers/blood/chemo
-	name = "\improper Phalanximine IV kit"
+	name = "\improper phalanximine IV kit"
 	desc = "IV kit for chemotherapy."
-	icon = 'icons/obj/chemopack.dmi'
+
+//if cancer gets re-introduced, someone will need to update on_reagent_change()
+//else&if you are able to make chemo via chemistry, chemopacks will be seen as "murky bloodpacks"
+//(unless they are being spawned in the map at roundstart like the other common bloodpacks)
+/////////////////////////////////////////
+//not needed anymore
+//	icon = 'icons/obj/chemopack.dmi'
+////////////////////////////////////////
 	New()
 		..()
 		reagents.clear_reagents()
 		reagents.add_reagent(PHALANXIMINE, 200)
 		update_icon()
+
+/obj/item/weapon/reagent_containers/blood/attackby(obj/item/W as obj, mob/user as mob)
+	var/turf/T = get_turf(src)
+	var/datum/reagent/blood/B = locate(/datum/reagent/blood) in reagents.reagent_list
+
+	if(mode == BLOODPACK_CUT)
+		return
+
+	if(istype(W, /obj/item/weapon/reagent_containers/syringe))
+		var/datum/reagent/blood/S = locate(/datum/reagent/blood) in W.reagents.reagent_list
+		//if a syringe with infected blood is used, it infects the blood inside the bloodpack.
+		if (S != null && S.data["virus2"] && reagents.has_reagent(BLOOD))
+			var/list/virus = B.data["virus2"]
+			virus |= virus_copylist(S.data["virus2"])
+		return //stops from punching holes
+
+	if (W.sharpness_flags & (SHARP_BLADE|SHARP_TIP))
```

Comment:
I added it but I don't think it makes much of a difference, it looks like an extra thing added on top of sharp tips and/or blades.